addons/movments_accounts_balances/controllers/controllers.py

```diff
@@ -199,6 +199,194 @@ def delete_bill(self, bill_id):
         except Exception as e:
             return "Failed to delete bill: {}".format(e)
 
+    ##create/get/delete_bills payment
+    @api.model
+    def create_bill_payment(self, bill_ids, journal_id, payment_date, payment_method_line_id):
+        # Ensure the bills exist and are valid
+        AccountMove = self.env['account.move']
+        bills = AccountMove.browse(bill_ids)
+        if not bills.exists():
+            raise ValueError("No bills found with the provided IDs")
+
+        # Calculate total amount to be paid
+        total_amount = sum(bill.amount_residual for bill in bills)
+
+        # Create a context dictionary mimicking what the wizard would receive
+        payment_context = {
+            'active_model': 'account.move',
+            'active_id': bill_ids[0],
+            'active_ids': bill_ids,
+        }
+
+        # Initialize the payment register wizard with the context
+        PaymentRegister = self.env['account.payment.register'].with_context(**payment_context).create({
+            'payment_date': payment_date,
+            'journal_id': journal_id,
+            'payment_method_line_id': payment_method_line_id,
+            'amount': total_amount,
+        })
+
+        # Simulate the 'action_create_payments' which is typically called when the wizard is submitted
+        payment = PaymentRegister.action_create_payments()
+
+        # Since 'action_create_payments' doesn't return the payment record, you may need to fetch it
+        # This assumes the last created payment is the one we're interested in, which might not always be correct in high concurrency situations.
+        # A more robust approach would be to analyze the payment creation logic to accurately fetch the related payment record.
+        last_payment = self.env['account.payment'].search([], order='create_date desc', limit=1)
```

Comment:
Can't you search reliably for the bill payment for the bills that were passed in, so this is reliable instead of looking for last payment?